librz/core/cmd/cmd_analysis.c

```diff
@@ -8812,3 +8443,220 @@ RZ_IPI RzCmdStatus rz_analysis_class_vtable_lookup_handler(RzCore *core, int arg
+			pj_kn(pj, "size", block->size);
+			if (block->jump != UT64_MAX) {
+				pj_kn(pj, "jump", block->jump);
+			}
+			if (block->fail != UT64_MAX) {
+				pj_kn(pj, "fail", block->fail);
+			}
+			if (xrefs) {
+				pj_ka(pj, "xrefs");
+				RzListIter *iter2;
+				ut64 *addr;
+				rz_list_foreach (xrefs, iter2, addr) {
+					pj_n(pj, *addr);
+				}
+				pj_end(pj);
+			}
+			if (calls) {
+				pj_ka(pj, "calls");
+				RzListIter *iter2;
+				ut64 *addr;
+				rz_list_foreach (calls, iter2, addr) {
+					pj_n(pj, *addr);
+				}
+				pj_end(pj);
+			}
+			pj_ka(pj, "fcns");
+			RzListIter *iter2;
+			RzAnalysisFunction *fcn;
+			rz_list_foreach (block->fcns, iter2, fcn) {
+				pj_n(pj, fcn->addr);
+			}
+			pj_end(pj);
+			pj_end(pj);
+			break;
+		case RZ_OUTPUT_MODE_TABLE: {
+			char *jump = block->jump != UT64_MAX ? rz_str_newf("0x%08" PFMT64x, block->jump) : strdup("");
+			char *fail = block->fail != UT64_MAX ? rz_str_newf("0x%08" PFMT64x, block->fail) : strdup("");
+			char *call = ut64join(calls);
+			char *xref = ut64join(calls);
+			char *fcns = fcnjoin(block->fcns);
+			rz_table_add_rowf(table, "xnddsssss",
+				block->addr,
+				block->size,
+				block->traced,
+				block->ninstr,
+				jump,
+				fail,
+				fcns,
+				call,
+				xref);
+			free(jump);
+			free(fail);
+			free(call);
+			free(xref);
+			free(fcns);
+		} break;
+		case RZ_OUTPUT_MODE_QUIET:
+			rz_cons_printf("0x%08" PFMT64x "\n", block->addr);
+			break;
+		default:
+			rz_cons_printf("0x%08" PFMT64x, block->addr);
+			if (block->jump != UT64_MAX) {
+				rz_cons_printf(" .j 0x%08" PFMT64x, block->jump);
+			}
+			if (block->fail != UT64_MAX) {
+				rz_cons_printf(" .f 0x%08" PFMT64x, block->fail);
+			}
+			if (xrefs) {
+				RzListIter *iter2;
+				rz_cons_printf(" .x");
+				ut64 *addr;
+				rz_list_foreach (xrefs, iter2, addr) {
+					rz_cons_printf(" 0x%08" PFMT64x, *addr);
+				}
+			}
+			if (calls) {
+				rz_cons_printf(" .c");
+				RzListIter *iter2;
+				ut64 *addr;
+				rz_list_foreach (calls, iter2, addr) {
+					rz_cons_printf(" 0x%08" PFMT64x, *addr);
+				}
+			}
+			if (block->fcns) {
+				RzListIter *iter2;
+				RzAnalysisFunction *fcn;
+				rz_list_foreach (block->fcns, iter2, fcn) {
+					rz_cons_printf(" .u 0x%" PFMT64x, fcn->addr);
+				}
+			}
+			rz_cons_printf(" .s %" PFMT64d "\n", block->size);
+		}
+		rz_list_free(xrefs);
+		rz_list_free(calls);
+	}
+	if (pj) {
+		pj_end(pj);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_basic_block_find_paths_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	ut64 addr = rz_num_math(core->num, argv[1]);
+	RzAnalysisBlock *block = rz_analysis_get_block_at(core->analysis, core->offset);
+	if (!block) {
+		RZ_LOG_ERROR("No basic block at 0x%" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzList *path = rz_analysis_block_shortest_path(block, addr);
+	if (!path) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	PJ *pj = state->d.pj;
```

Comment:
You should check the state mode, not whether null or not because d is a union and could contain other things

Reply:
I suggest to just use again cmd_state_outpit_start_array